File: sentimental-cash/cash.py
# TODO


def main():
    coins = 0
    while True:
        try:
            change_owed = float(input("Change owed: "))

            # if not grater than 0 re-prompt again
            if change_owed > 0:
                cents = round(change_owed * 100)

                # loop
                while cents > 0:
                    # quarters
                    if cents >= 25:
                        cents -= 25
                        coins += 1

                    # dimes
                    elif cents >= 10:
                        cents -= 10
                        coins += 1

                    # nickels
                    elif cents >= 5:
                        cents -= 5
                        coins += 1

                    # pennies
                    elif cents >= 1:
                        cents -= 1
                        coins += 1

                print(coins)
                break  # exit loop

        except:
            continue


if __name__ == "__main__":
    main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("Change owed: %i\n", coins);
}

// get user input
int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);
    return cents;
}

// if cents == 25,cents - 25, quarters += 1
int calculate_quarters(int cents)
{
    int quarters;
    quarters = 0;
    while (cents >= 25)
    {
        cents = cents - 25;
        quarters++;
    }
    return quarters;
}
// if cents >= 10, cents - 10, dimes += 1
int calculate_dimes(int cents)
{
    int dimes;
    dimes = 0;
    while (cents >= 10)
    {
        cents = cents - 10;
        dimes++;
    }
    return dimes;
}

// if cents >= 5,cents - 5, nickels += 1
int calculate_nickels(int cents)
{
    int nickels;
    nickels = 0;
    while (cents >= 5)
    {
        cents = cents - 5;
        nickels++;
    }
    return nickels;
}

// if cents >= 1, cents - 1, pennies += 1
int calculate_pennies(int cents)
{
    int pennies;
    pennies = 0;
    while (cents >= 1)
    {
        cents = cents - 1;
        pennies++;
    }
    return pennies;
}
